### PR TITLE
Fix detecting strings with hashtag as requiring quotes

### LINKED ISSUE #271: BUG: quoted-strings only-when-needed with hashtag
Setup
---

_test.yml_:
```yaml
---
hashtag: 'Test #hashtag'
```

_.yamllint.yml_:  
```yaml
extends: default

rules:
  quoted-strings:
    required: only-when-needed
```

Result
---

```
$ yamllint test.yml 
test.yml
  2:10      error    string value is redundantly quoted with any quotes  (quoted-strings)
```

### PR DESCRIPTION
**Pull request by julienfalque**
Fixes #271.

This is my first Python contribution ever, please let me know if there's anything wrong :)

### CODE REVIEW

File: yamllint/rules/quoted_strings.py [hunk 1]
```diff
@@ -160,7 +160,8 @@ def _quotes_are_needed(string):
     try:
         a, b = loader.get_token(), loader.get_token()
         if (isinstance(a, yaml.ScalarToken) and a.style is None and
-                isinstance(b, yaml.BlockEndToken)):
+                isinstance(b, yaml.BlockEndToken) and
+                a.value == string):
```
Comment (adrienverge) [inline yamllint/rules/quoted_strings.py:164]: Can you merge these two lines?

**Review by adrienverge — COMMENTED** [review]
Looks good. Does anyone see any corner cases that would fail with this change?

> This is my first Python contribution ever

Congrats, it's short but neat!

File: yamllint/rules/quoted_strings.py (hunk 1 above)
Comment (julienfalque) [reply yamllint/rules/quoted_strings.py]: Done.

**Comment by adrienverge** [comment]
> Note: Travis CI builds do run as expected but don't seem to be reported to Github. Are the repository settings correct?

It's a know bug of theirs (I've seen it on many repos). I migrated the repo from travis-ci.org to travis-ci.com, it should work now. Can you `push --force` to the branch to trigger a new build? Thanks!

**Comment by julienfalque** [comment]
Looking good but it seems the old integration is still enabled :)

**Pull request merged**